### ISSUE TITLE
FlagsConstants objects need __nonzero__

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6303]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)| @wsanchez reported|
|-|-|
|Trac ID|trac#6303|
|Type|enhancement|
|Created|2013-02-14 21:41:32Z|
|Branch|https://github.com/twisted/twisted/tree/flags-nonzero-6303|

This is rather unexpected:

```
>>> bool(QueryFlags.caseInsensitive & QueryFlags.NOT)
True
```

You have to use `.value`:

```
>>> bool((QueryFlags.caseInsensitive & QueryFlags.NOT).value)
False
>>> bool((QueryFlags.caseInsensitive & QueryFlags.caseInsensitive).value)
True
```

<details><summary>Searchable metadata</summary>

```
trac-id__6303 6303
type__enhancement enhancement
reporter__wsanchez wsanchez
priority__normal normal
milestone__ 
branch__branches_flags_nonzero_6303 branches/flags-nonzero-6303
branch_author__wsanchez wsanchez
status__closed closed
resolution__duplicate duplicate
component__core core
keywords__ 
time__1360878092000000 1360878092000000
changetime__1370043781000000 1370043781000000
version__None None
owner__wsanchez wsanchez

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_1"></a>|@wsanchez commented|
|-|-|

(In [37141]) Branch for [#6303](https://github.com/twisted/twisted/issues/6303)

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_2"></a>|@wsanchez commented|
|-|-|

An updated fix for is included with [#6302](https://github.com/twisted/twisted/issues/6302).

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)|@wsanchez set owner to @wsanchez<br>@wsanchez set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_4"></a>|@wsanchez set status to `closed`|
|-|-|

With was fixed with [#6302](https://github.com/twisted/twisted/issues/6302)
